science/gnss-sdr/Portfile

```diff
@@ -74,14 +74,23 @@ depends_build-append \
 depends_lib-append  \
     port:armadillo \
     port:google-glog \
-    port:lapack \
     port:matio \
     port:protobuf3-cpp \
     port:pugixml \
     path:lib/libgnuradio-runtime.dylib:gnuradio \
     path:lib/libuhd.dylib:uhd \
     path:lib/pkgconfig/gnutls.pc:gnutls
 
+# NOTE: gnss-sdr avoided using the Accelerate framework,
+# since it was buggy in previous macOS versions.
+# This changes in Sonoma, where the bugs seems gone,
+# and gnss-sdr do link against it (making the lapack port unnecesary)
+
+if {${os.major} <= 22} {
```

Comment:
this version check is only valid on macOS so it should include a check for that as well, per our usual way:
"if {${os.platform} eq "darwin" &&${os.major} <= 22} {`

Reply:
also, what happens if someone has `lapack` installed on `${os.major} >22`? To get reproducible builds one needs to ensure that it will not link to it with your current approach, as otherwise uninstalling `lapack` will break things. So it seems that this will need a little bit more work.

Reply:
If someone has  `lapack` installed on `${os.major} >22` the port will ignore it. Honestly, this was done because the `lapack`port was broken in Sonoma, but now it seems to be fixed. In order to do not mess up things, I will include back the `lapack` port as a requirement also in Sonoma, so all OS versions will have the same list of dependencies. The port should equally build.

Reply:
if it's ignored on > 22 even if it's installed then please don't add it as a unnecessary dependency. The check you had is fine, just please leave add a comment to state explain it.

Reply:
ok, I removed the lapack port as a dependency in Sonoma (so is only required if ```{${os.platform} eq "darwin" && ${os.major} <= 22}```), and added a comment about it.